BSP/Simulator/Src/BSP_CAN.c

```diff
@@ -124,6 +126,26 @@ uint8_t BSP_CAN_Read(CAN_t bus, uint32_t* id, uint8_t* data) {
         data[i] = (fullData >> (8 * (len-i-1))) & 0xFF;
```

Comment:
Initialize _len_ variable to 0. 
It seems to happen that since the line we are reading form the csv file in testCAN3() is empty, _len_ doesn't get initialized to 0 with sscanf from the char array currentCAN but a random number. This causes conflicts with this for loop running a random number of times (depending on what value _len_ got randomly initizialized to) because the _data_ array's size might be less that _len_.